pagefaultgames/pokerogue
# [Bug] Let Shedinja relearn egg moves when evolved from a starter

## What are the changes the user will see?
After evolution, Shedinja's "met" biome and level will match that of the Nincada it evolved from. This will allow Shedinja to use Memory Mushrooms to relearn any egg moves the user has unlocked for Nincada, if Shedinja evolved from a starter Nincada.

## Why am I making these changes?
https://github.com/pagefaultgames/pokerogue/issues/3842
This change makes Shedinja's relearn behavior the same as any other Pokemon evolved from a starter.

## What are the changes from a developer perspective?
When Nincada evolves, the evolution is handled specially by creating a second Nincada in the user's party and setting many of its attributes to be the same as the original Nincada's. This PR makes it so the second Nincada also copies metadata about where and when the original one was met.

### Screenshots/Videos
Nincada before evolution:
![Screen Shot 2024-08-31 at 12 23 14 PM](https://github.com/user-attachments/assets/dd171557-b385-4ae0-b0f1-13bdae540215)

Shedinja being able to learn an egg move after evolution:

https://github.com/user-attachments/assets/811b518a-d71f-4e2f-820b-c52f6f358d0f

## How to test the changes?
 - Automated: Added a `metBiome` check in the Shedinja evolution test.
 - Manual: Evolve a Nincada under the standard conditions that lead to Shedinja (at least one empty spot in party and Pokeball in the bag) and verify that the Shedinja's met information on its summary screen matches what Nincada's was.

## Checklist
